backend/api/Services/AssetDeckService.cs

```diff
@@ -49,20 +57,58 @@ private IQueryable<AssetDeck> GetAssetDecks()
                 .FirstOrDefaultAsync(assetDeck => assetDeck.Id.Equals(id));
         }
 
-        public async Task<AssetDeck> Create(CreateAssetDeckQuery newAssetDeck)
+        public async Task<IEnumerable<AssetDeck>> ReadByAsset(string asset)
+        {
+            return await _context.AssetDecks.Where(a =>
+                a.AssetCode.Equals(asset)).Include(a => a.SafePositions).ToListAsync();
+        }
+
+        public async Task<AssetDeck?> ReadByAssetAndDeck(string asset, string deck)
+        {
+            return await _context.AssetDecks.Where(a =>
+                a.AssetCode.Equals(asset) &&
+                a.DeckName.Equals(deck)
+            ).Include(a => a.SafePositions).FirstOrDefaultAsync();
+        }
+
+        public async Task<AssetDeck> Create(CreateAssetDeckQuery newAssetDeck, List<Pose> safePositions)
         {
+            var sp = new List<SafePosition>();
+            foreach (var p in safePositions)
+            {
+                sp.Add(new SafePosition(p));
+            }
             var assetDeck = new AssetDeck
             {
                 AssetCode = newAssetDeck.AssetCode,
                 DeckName = newAssetDeck.DeckName,
-                DefaultLocalizationPose = newAssetDeck.DefaultLocalizationPose
+                DefaultLocalizationPose = newAssetDeck.DefaultLocalizationPose,
+                SafePositions = sp
             };
-
             await _context.AssetDecks.AddAsync(assetDeck);
             await _context.SaveChangesAsync();
             return assetDeck;
         }
 
+        public async Task<AssetDeck> Create(CreateAssetDeckQuery newAssetDeck)
+        {
+            var assetDeck = await Create(newAssetDeck, new List<Pose>());
+            return assetDeck;
+        }
+
+        public async Task<AssetDeck?> AddSafePosition(string asset, string deck, SafePosition safePosition)
+        {
+            var assetDeck = await ReadByAssetAndDeck(asset, deck);
+            if (assetDeck is null)
+            {
+                return null;
+            }
+            assetDeck.SafePositions.Add(safePosition);
```

Comment:
Have you tested this if the asset deck has no list to begin with? 
I think you might have to initialize the list in that case

Reply:
In the initialization of an AssetDeck, an empty list is made, so you are able to use this service even if there are no SafePositions prior

Reply:
Have you tested it? Might behave differently than expected due to C# -> Database -> C#

Reply:
Tested it locally, with SQLite.